homeassistant/components/digital_ocean/__init__.py

```diff
@@ -81,6 +93,45 @@ def get_droplet_id(self, droplet_name):
 
         return droplet_id
 
+    @Throttle(MIN_TIME_BETWEEN_DOMAIN_UPDATES)
+    def update_domain_record(
```

Comment:
Why is half the service implemented in a separate module?
Please either move all of the implementation to services.py, or move it all to here.

Reply:
It was already like that. The code for this integration is pretty poorly organized, and this PR is already long as it is. I really don't want to mess with moving code/refactoring here.

I'll keep maintaining the code (organizing, adding config flow) in upcoming PRs.

Reply:
I don't understand this comment, your PR adds a new file `service.py`, how can you say it was already like that?

Reply:
The idea is to BEGIN that separation with this PR, instead of just mashing it up on that same file.
In other words, pave the way for a more modular integration

Reply:
OK, but in that case, why do you add the method `DigitalOcean.update_domain_record`?
It seems the only think the method needs is access to `DigitalOcean._access_token`, you could just unprotect that?